### PR TITLE
feat(wrecks): editable wreck catalogue with site and feature links

### PR DESCRIPTION
**Pull request by ericgriffin**
## Summary

Slice 3a of the seascape usefulness program: an editable wreck catalogue, parallel to dive sites.

Divers know things about wrecks that no site record can hold: what the vessel was, how deep the deck sits versus the seabed, when and how it sank, whether it is intact enough to penetrate, whether it is protected. Until now the only place for any of that was a site's free-text notes, or the `wreck` marker slice 2 added, which carries a name and a position and nothing else. A wreck also outlives any one site: you can know about it long before you log a site for it.

- **`wrecks` table (schema v153)**: a full LWW entity with its own HLC, mirroring `DiveSites` structurally (nullable `diverId`, `isShared`) so ownership, sharing, and sync behave identically. Position is optional (a wreck can be known before its coordinates are), and `siteId` is a nullable link that **NULLs rather than cascades** on site delete, because a wreck must outlive the diver's decision to stop logging a site there.
- **`site_features` gains `wreckId` and `source`**: a wreck marker can now point at the catalogue record instead of duplicating its name and depth. The `source` column defaults to `'diver'` and exists for slice 3b, added in the same migration to avoid a second schema pass.
- **Sync**: `wrecks` is enrolled as a conflict-capable entity across the serializer's twelve points, the merge order, `entityHasUpdatedAt`, `_hlcTargets`, and `parentRefs` (both FKs nullable). The `siteFeatures` parent refs gain `wreckId`. The delta export filters on the row's own HLC.
- **Site merge**: linked wrecks re-point to the survivor inside the existing transaction, and `MergeSnapshot` carries their prior `siteId` so undo restores them.
- **UI**: wrecks become the 16th nav destination (`/wrecks`, unpinned, so it appears in the wide rail and the phone overflow sheet). Three pages reuse the sites feature's master-detail scaffold rather than duplicating it: a searchable list, a detail page showing the structured facts in the diver's unit with the linked site, and an edit form. Site detail gains a "Wrecks here" section, and the feature edit sheet gains a wreck picker when the type is `wreck`.

## Notes for review

- Enum-valued columns store the raw enum NAME and the entity keeps the raw string beside a nullable typed getter, so a value from a newer build round-trips through sync instead of being dropped. Tests cover that for both the entity and the repository.
- The sheet's wreck picker resolves its list by AWAITING the provider future rather than reading it: the same `ref.read` shape silently dropped taps in slice 2 until the coverage pass caught it.
- Adding a nav destination shifts six test contracts (destination count and id order, movable ids, rail order, the overflow provider, and two index/count assertions in the scaffold suite). All are updated here; the accent palette needed entries in both the light and dark maps.
- Explicitly out of scope: external sources (OSM Overpass and NOAA ENC are slice 3b, which writes into this table), wreck-specific 3D rendering (3c), wreck merge, and photos.

## Testing

- Unit: schema round-trip including the `setNull` behaviors both ways, domain entity (unknown enum names, `hasCoordinates`, the three clear flags), repository CRUD with the write ritual asserted (pending mark, HLC stamp, no parent bump), sync round-trip plus own-HLC delta filtering, and merge relink and undo.
- Widget: list search and empty state, detail rendering in both unit systems and with an unknown enum name, edit create/update/validation with feet-to-meters conversion, the site detail section, and the wreck picker appearing only for the wreck type and dropping its link when the type changes.
- Full suite green, `flutter analyze` clean, `dart format` clean, no l10n drift (39 keys across all 11 locales).

### CODE REVIEW

File: lib/features/site_scape/presentation/site_feature_sheet.dart [hunk 1]
```diff
@@ -200,6 +226,26 @@ class _SiteFeatureSheetState extends ConsumerState<SiteFeatureSheet> {
...
+          // Only a wreck marker can point at a catalogue wreck.
+          if (_typeName == SiteFeatureType.wreck.name) ...[
+            const SizedBox(height: 12),
+            DropdownButtonFormField<String?>(
+              key: const ValueKey('siteFeatureWreckField'),
+              initialValue: _wrecks.any((w) => w.id == _wreckId)
+                  ? _wreckId
+                  : null,
+              decoration: InputDecoration(labelText: l10n.wrecks_field_site),
+              items: [
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/site_scape/presentation/site_feature_sheet.dart:239]: Fixed in aaf9cb44bbf: the picker now uses wrecks_link ("Link a wreck"), which already existed and says exactly what the control does.

File: lib/features/wrecks/presentation/pages/wreck_edit_page.dart [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+            const SizedBox(height: 12),
+            Row(
+              children: [
+                Expanded(
+                  child: _numberField(
+                    key: 'wreckLatitudeField',
+                    controller: _latitude,
+                    label: 'Lat',
+                  ),
+                ),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/wrecks/presentation/pages/wreck_edit_page.dart:283]: Fixed in aaf9cb44bbf: added wrecks_field_latitude and wrecks_field_longitude across all 11 locales rather than borrowing the diveSites/diveCenters keys, so the wrecks feature owns its own strings.

File: lib/features/wrecks/presentation/pages/wreck_edit_page.dart [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+        signed: true,
+      ),
+      decoration: InputDecoration(labelText: label),
+    );
+  }
+
+  Future<void> _save(double unitInMeters) async {
+    if (!(_formKey.currentState?.validate() ?? false)) return;
+    final lat = double.tryParse(_latitude.text.trim());
+    final lon = double.tryParse(_longitude.text.trim());
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/wrecks/presentation/pages/wreck_edit_page.dart]: Fixed in aaf9cb44bbf: both coordinates now go through the same _number helper the depth and length fields use, so "12,15" parses. Added a test entering comma coordinates and asserting the stored doubles (it needed scrollUntilVisible, since the coordinate fields sit far down a lazily built ListView).

File: lib/features/wrecks/presentation/widgets/site_wrecks_section.dart [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                subtitle: Text(
+                  [
+                    wreckVesselTypeLabel(l10n, w.vesselTypeName),
+                    wreckMeasure(
+                      w.depthToDeckMeters,
+                      unitInMeters,
+                      depthUnit.symbol,
+                    ),
+                  ].where((s) => s.isNotEmpty).join(' • '),
+                ),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/wrecks/presentation/widgets/site_wrecks_section.dart]: Fixed in aaf9cb44bbf: the subtitle is computed first and passed as null when empty, matching what the catalogue list already did. Added a test asserting subtitle is null for a wreck with no type and no depth.

File: lib/features/wrecks/presentation/pages/wreck_edit_page.dart [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+        child: ListView(
+          padding: const EdgeInsets.all(16),
+          children: [
+            TextFormField(
+              key: const ValueKey('wreckNameField'),
+              controller: _name,
+              decoration: InputDecoration(labelText: l10n.wrecks_field_name),
+              validator: (v) =>
+                  (v == null || v.trim().isEmpty) ? l10n.wrecks_add : null,
+            ),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/wrecks/presentation/pages/wreck_edit_page.dart:142]: Fixed in aaf9cb44bbf: added wrecks_validation_nameRequired ("Enter a name") in all 11 locales; the validator no longer reuses the action label.

**Comment by ericgriffin** [comment]
Will revisit later
